pkg/token/service.go

```diff
@@ -423,72 +438,105 @@ func (d *tokenService) updateToken(key CacheKey, tt mode) error {
 	}
 }
 
-func (ts *tokenService) writeFilesWithRetry(ctx context.Context, maxElapsedTime time.Duration) error {
-	// backoff config with first retry delay of 5s, and then 10s, 20s, ...
-	b := backoff.NewExponentialBackOff()
-	b.InitialInterval = 5 * time.Second
-	b.Multiplier = 2
-	b.MaxElapsedTime = maxElapsedTime
-
-	return backoff.RetryNotify(ts.writeFiles, backoff.WithContext(b, ctx), func(err error, backoffDelay time.Duration) {
-		log.Errorf("Failed to write token files: %s. Retrying in %s", err.Error(), backoffDelay)
-	})
-}
-
-func (d *tokenService) writeFiles() error {
+func (d *tokenService) writeFiles(ctx context.Context, maxElapsedTime time.Duration) []error {
```

Comment:
I understand you have renamed the function name `writeFilesWithRetry` to `writeFiles` by making retries default behavior for the writing files.

But again this is not a good thing to refactor in a PR for a new implementation of `atTargetDomainRoles` &  `rtTargetDomainRoles`
You can write a simple comment TODOs saying
Refactoring must be done in a different PR.